### PR TITLE
chore: bumping vite to latest so we can test vite-dev-server changes

### PR DESCRIPTION
**Pull request by JessicaSachs**
This PR bumps Vite to 2.8.4 in preparation for fixing our test infra.

### CODE REVIEW

**Review by BlueWinds — DISMISSED** [review]

File: yarn.lock [hunk 1]
```diff
@@ -19668,11 +19668,131 @@ es6-weak-map@^2.0.1:
     es6-iterator "^2.0.3"
     es6-symbol "^3.1.1"
 
-esbuild@^0.12.17, esbuild@^0.12.28:
+esbuild-android-arm64@0.14.23:
```
Comment (BlueWinds) [inline yarn.lock:19671]: Interesting that latest vite broke this out into so many pieces.

**Comment by BlueWinds** [comment]
I also see a failing test that looks related - https://app.circleci.com/pipelines/github/cypress-io/cypress/32751/workflows/a0f9d81c-fa23-4086-a3bf-c32fa45ba3e4/jobs/1289370

**Comment by JessicaSachs** [comment]
> I also see a failing test that looks related - [app.circleci.com/pipelines/github/cypress-io/cypress/32751/workflows/a0f9d81c-fa23-4086-a3bf-c32fa45ba3e4/jobs/1289370](https://app.circleci.com/pipelines/github/cypress-io/cypress/32751/workflows/a0f9d81c-fa23-4086-a3bf-c32fa45ba3e4/jobs/1289370)

Yeah, it was. Mismatch in plugin versions, vue versions, etc. I just bumped everything in the repo to be the same version.

**Review by lmiller1990 — DISMISSED** [review]
Code seems fine, just need to wait for CI

**Review by ImCesar — DISMISSED** [review]

**Comment by BlueWinds** [comment]
Now we have typecheck failures. ^^;

**Comment by JessicaSachs** [comment]
I'm not going to merge this PR until I get the Vite Dev Server refactor in. The latest Vite server + our current Vite Dev Server is really flaky. Most of the Vite-based test runs fail ~ >50% of the time. When you restart the server, they may/may not flake, so it's pretty difficult to isolate this issue.

After going through the effort of debugging this flake, I realize that I have to update some plugins when we bump Vite internally to get everything to work.

I'm gonna do some of those now and then focus on cutting over the server so that it'll work with the newer versions of Vite.

**Pull request closed without merging**